code/modules/mob/living/simple_animal/abnormality/he/pinocchio.dm

```diff
@@ -148,6 +145,20 @@
 	ADD_TRAIT(realboy, TRAIT_NODROP, "Abnormality")
 	realboy.update_icon()
 
+	//--Side Gamemodes stuff--
+	//For playable abnormalities, directly lets the playing currently controlling pinocchio get control of the spawned mob
+	if(client)
+		mind.transfer_to(realboy)
+	//Prevents pinocchio from going insane on Rcorp.
+	if(!CheckCombat())
+		realboy.ai_controller = /datum/ai_controller/insane/murder/puppet
+		realboy.InitializeAIController()
+		realboy.apply_status_effect(/datum/status_effect/panicked_type/puppet)
```

Comment:
Should be an else if, I believe. Right now it will just force AI controller on it even if it is controlled by a player(on normal maps).

Reply:
With the flashlight spawn below, however, you will have to account for it, so a pure else if won't work here.

Reply:
> Should be an else if, I believe. Right now it will just force AI controller on it even if it is controlled by a player(on normal maps).

That's actually intended! The AI controller now gets disabled when someone is inside the mob, so it keeps the AI if the player ghosts for some reason while playing with the mob on playables. We dont want the insanity on Rcorp though as it would friendly fire hit other abnormalities.